src/NodeTypeResolver/NodeTypeResolver/NameTypeResolver.php

```diff
@@ -37,6 +37,11 @@ public function getNodeClasses(): array
      */
     public function resolve(Node $node): Type
     {
+        // not instanceof FullyQualified means it is a Name
+        if (! $node instanceof FullyQualified && $node->hasAttribute(AttributeKey::NAMESPACED_NAME)) {
+            return $this->resolve(new FullyQualified($node->getAttribute(AttributeKey::NAMESPACED_NAME)));
+        }
```

Comment:
this is really needed to patch `StaticCall` that imported fixture https://github.com/rectorphp/rector-src/pull/6236/files#diff-ec6cd31df51e19b0199b350a318339fbc3e2b1aa740593511a466849e7fc5289 as a proof that `namespacedName` attribute is needed on `NameTypeResolver` 👍